dataset/create_twitter_dataset.py

```diff
@@ -0,0 +1,199 @@
+                    #   2) A writes to his own timeline
+                    #   3) A reads his own timeline
+                    access_timeline(timestamp, first_uid, first_user_ip, first_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+                    result_file.write(LOG_FORMAT % (timestamp, first_uid, first_user_ip, 'null', first_target_server, 'READ', '200', '2800'))
+                    result_file.write(LOG_FORMAT % (timestamp, tweet_uid, first_user_ip, 'null', first_target_server, 'WRITE', '201', '0'))
+                    tweet_server[tweet_uid] = first_target_server  # must be here, because this is were the tweet is already stored on the server.
+                    if first_uid not in user_timeline_map:  # add the tweet to the timeline
+                        user_timeline_map[first_uid] = []
+                    elif len(user_timeline_map[first_uid]) > TIMELINE_LIMIT:
+                        user_timeline_map[first_uid].pop(0)  # remove the first element
+                    user_timeline_map[first_uid].append(tweet_uid)
+                    access_timeline(timestamp, first_uid, first_user_ip, first_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+
+            elif mode == TWITTER_RT or mode == TWITTER_RE:
+                # Five steps: assuming A is the sender of the tweet and B is the recipient of the tweet.
+                #   1) A reads B's timeline
+                #   2) A reads his own timeline
+                #   3) A writes the tweet to A's timeline
+                #   4) A reads B's timeline (as an effect of retweeting and replying)
+                #   5) A reads A's timeline
+                access_timeline(timestamp, second_uid, first_user_ip, second_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+                access_timeline(timestamp, first_uid, first_user_ip, first_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+                result_file.write(LOG_FORMAT % (timestamp, tweet_uid, first_user_ip, 'null', first_target_server, 'WRITE', '201', '0'))
+                tweet_server[tweet_uid] = first_target_server  # must be here, because this is were the tweet is already stored on the server.
+                if first_uid not in user_timeline_map:  # add the tweet to the timeline
+                    user_timeline_map[first_uid] = []
+                elif len(user_timeline_map[first_uid]) > TIMELINE_LIMIT:
+                    user_timeline_map[first_uid].pop(0)  # remove the first element
+                user_timeline_map[first_uid].append(tweet_uid)  # add the tweet to the timeline
+                if mode == TWITTER_RE:
+                    reply_stream_map[tweet_uid] = second_uid # the tweet is directed to the second_uid
+
+                result_file.write(LOG_FORMAT % (timestamp, second_uid, first_user_ip, tweet_uid, second_target_server, 'READ', '200', str(TWEET_SIZE)))
```

Comment:
I had wondered whether the `source` should be `first_target_server` instead of `first_user_ip`, because I was perhaps imagining that the server that the tweet is posted on is making a remote request to the other user's tweets instead of the original client. Thoughts?